requirements.txt

```diff
@@ -28,3 +28,4 @@
 numpy>=1.24.0
 pandas>=2.1.0
 protobuf>=4.21.1
+pyarrow
```

Comment:
Could a version specifier be added to this dependency?

Reply:
@oliviawebber , any suggestions for `pyarrow` versions? I don't want to introduce incompatibilities with other packages.

Reply:
If there's no known minimum, my usual rule of thumb is to pick a version from like ~1 year ago which would give `14.0.0`. The one library I know we use that might eventually pull in `pyarrow` looks like it requires `>= 7.0.0`.

Reply:
Sorta related: I ran into this issue https://stackoverflow.com/questions/78634235/numpy-dtype-size-changed-may-indicate-binary-incompatibility-expected-96-from recently. Based on that (and maybe this migration guide https://numpy.org/devdocs/numpy_2_0_migration_guide.html) do we also need to pin to `numpy<2.0.0`?

Reply:
Yes, I am using numpy C interface that may need porting to 2. So will have
```
numpy>=1.25,<2.0.0
pyarrow>=14.0.0
```